File: CS/E689.Module/DatabaseUpdate/Updater.cs
using System;
using System.Linq;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Updating;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Security;
//using DevExpress.ExpressApp.Reports;
//using DevExpress.ExpressApp.PivotChart;
//using DevExpress.ExpressApp.Security.Strategy;
//using E689.Module.BusinessObjects;

namespace E689.Module.DatabaseUpdate {
    // For more typical usage scenarios, be sure to check out http://documentation.devexpress.com/#Xaf/clsDevExpressExpressAppUpdatingModuleUpdatertopic
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) :
            base(objectSpace, currentDBVersion) {
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            string name = "Demo Object with a FlagsAttribute Enumeration Property";
            DemoObject theObject = ObjectSpace.FindObject<DemoObject>(CriteriaOperator.Parse("Name=?", name));
            if(theObject == null) {
                theObject = ObjectSpace.CreateObject<DemoObject>();
                theObject.Name = name;
                theObject.TestMe = TestFlagsAttributeEnum.Air | TestFlagsAttributeEnum.Water;
            }
        }
    }
}

File: VB/E689.Module/DatabaseUpdate/Updater.vb
Imports System.Linq
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Security
'using DevExpress.ExpressApp.Reports;
'using DevExpress.ExpressApp.PivotChart;
'using DevExpress.ExpressApp.Security.Strategy;
'using E689.Module.BusinessObjects;

Namespace E689.Module.DatabaseUpdate
	' For more typical usage scenarios, be sure to check out http://documentation.devexpress.com/#Xaf/clsDevExpressExpressAppUpdatingModuleUpdatertopic
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			Dim name As String = "Demo Object with a FlagsAttribute Enumeration Property"
			Dim theObject As DemoObject = ObjectSpace.FindObject(Of DemoObject)(CriteriaOperator.Parse("Name=?", name))
			If theObject Is Nothing Then
				theObject = ObjectSpace.CreateObject(Of DemoObject)()
				theObject.Name = name
				theObject.TestMe = TestFlagsAttributeEnum.Air Or TestFlagsAttributeEnum.Water
			End If
		End Sub
	End Class
End Namespace
